lib/web_ui/lib/src/engine/scene_view.dart

```diff
@@ -305,23 +305,28 @@ final class PlatformViewContainer extends SliceContainer {
   @override
   void updateContents() {
     assert(_styling != null);
-    assert(_size != null);
+    assert(_bounds != null);
     if (_dirty) {
       final DomCSSStyleDeclaration style = container.style;
       final double devicePixelRatio = EngineFlutterDisplay.instance.devicePixelRatio;
-      final double logicalWidth = _size!.width / devicePixelRatio;
-      final double logicalHeight = _size!.height / devicePixelRatio;
+      final double logicalWidth = _bounds!.width / devicePixelRatio;
+      final double logicalHeight = _bounds!.height / devicePixelRatio;
       style.width = '${logicalWidth}px';
       style.height = '${logicalHeight}px';
       style.position = 'absolute';
 
-      final ui.Offset? offset = _styling!.position.offset;
-      final double logicalLeft = (offset?.dx ?? 0) / devicePixelRatio;
-      final double logicalTop = (offset?.dy ?? 0) / devicePixelRatio;
+      final PlatformViewPosition position = PlatformViewPosition.combine(
+        _styling!.position,
+        PlatformViewPosition.offset(_bounds!.topLeft),
+      );
+
+      final ui.Offset offset = position.offset ?? ui.Offset.zero;
+      final double logicalLeft = offset.dx / devicePixelRatio;
+      final double logicalTop = offset.dy / devicePixelRatio;
```

Comment:
Up to you, but local variable types are now optional. It might be good to take advantage of it in individual PRs so we don't have to do a huge type rewrite later.